### PR TITLE
proposal: start instances ordered by their defined tiers

### PR DESCRIPTION
**Pull request by foxriver76**
- more or less an implementation of #279, however, I think a separate class is overkill because it's very hard to outsource the whole instance logic which is done in main.js with processes and schedules, timeouts, etc. also with only tier logic there is nothing to persist because a full instance startup is only performed at controller start or `start all cli call`
- I would propose something like, logic adapters are Tier 1, adapters implementing an API or controlling devices are Tier 2, adapters which are not really usable without tier 2 adapters should be tier 3, e. g. vis is not usable when I am not able to control devices yet and states may be old
- Admin is always first because that's how it was until now

### CODE REVIEW

File: lib/tools.js [hunk 1]
```diff
@@ -2733,6 +2733,54 @@ function isLocalStatesDbServer(dbType, host, checkIfLocalOnly) {
...
 
+/**
+ * Get ordered instances according to tier level
+ *
+ * @param {object} objects - Objects DB
+ * @param {object} logger - logger object
+ * @param {string} [logPrefix] - prefix for logging
+ * @return {Promise<object[]>}
+ */
+async function getOrderedInstances(objects, logger, logPrefix='') {
```
Comment (Apollon77) [inline lib/tools.js:2744]: add ordering info in method name? I know "gteInstanesOrderedByStartPriority" is lengthy bt more clear

**Review by Apollon77 — COMMENTED** [review]
good ... only a small naming comment from my side.

**Comment by Apollon77** [comment]
In generall ok ... only question open for me is still whewre the "tier" values comes from? Really coming from the io-package? Or do we have a second "central source"/"the repo" that brings this and we add that top the io-package "On adapter install"

**Comment by foxriver76** [comment]
What would be the point of maintaining a central source from your point of view?

**Comment by Apollon77** [comment]
only "control over adapters that decide for themself to put themself into tier 1 :-) no idea if this is relevant ... maybe we should just "Trust" at the beginning ...

**Comment by foxriver76** [comment]
Hmm could auto checker inform at a stable increase if someone changed the tier in between? Should be possible.. hopefully not necessary

**Comment by Apollon77** [comment]
I think we trust at the beginning and see then

**Review by Apollon77 — APPROVED** [review]
LGTM

**Pull request merged**